index.js

```diff
@@ -108,6 +127,31 @@ function outputChangelog (argv, cb) {
   })
 }
 
+function getNextVersion (release) {
+  var newVersion = semver.inc(pkg.version, release.releaseAs)
+
+  if (argv.preRelease) {
+    // hack: if a feat is commited after two patches it wants to increase
+    // the prerelease tag instead of the minor and start from 0
+    // the next 2 lines and the if statement should fix this
+    var realVersion = semver.inc(pkg.version, release.releaseAs)
+    var prereleaseVersion = semver.inc(pkg.version, 'prerelease', argv.tag)
+
+    if (semver.gt(realVersion, prereleaseVersion)) {
+      newVersion = semver.inc(pkg.version, release.releaseAs)
+      newVersion = newVersion + '-' + argv.tag + '.0'
+
+      if (semver.gt(prereleaseVersion, newVersion)) {
```

Comment:
@stvnwrgs rather than a rewrite, mind adding a test for each of the branches in the logic.

If any of the branches aren't necessary, this will probably jump out at you as you write the tests.